### ISSUE TITLE
Fix request: AvoidIncrementOrDecrementForVolatileField false positive on "not"

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#500]
```
class IssueX {
     private volatile boolean initialized;

     protected boolean doProcess(Msg msg) {
        if (!initialized) { // <- false positive
          doInit();
          initialized = true;
        }
}
```
